core/lib/multivm/src/tracers/call_tracer/mod.rs

```diff
@@ -0,0 +1,42 @@
+use once_cell::sync::OnceCell;
+use std::sync::Arc;
+use zksync_types::vm_trace::Call;
+
+pub mod vm_latest;
+pub mod vm_virtual_blocks;
+
+#[derive(Debug, Clone, Default)]
+pub struct CallTracer {
+    stack: Vec<FarcallAndNearCallCount>,
+    result: Option<Arc<OnceCell<Vec<Call>>>>,
+}
+#[derive(Debug, Clone)]
+struct FarcallAndNearCallCount {
+    farcall: Call,
+    near_calls_after: usize,
+}
+
+impl CallTracer {
+    pub fn new(result: Arc<OnceCell<Vec<Call>>>) -> Self {
+        Self {
+            stack: vec![],
+            result: Some(result),
+        }
+    }
+
+    fn extract_result(&mut self) -> Vec<Call> {
+        std::mem::take(&mut self.stack)
+            .into_iter()
+            .map(|x| x.farcall)
+            .collect()
+    }
+
+    fn store_result(&mut self) {
+        if self.result.is_none() {
+            return;
+        }
```

Comment:
Is this valid to call `store_result` with `result` already being `Some`? E.g. won't we accidentally hide a potential bug by doing nothing here? 

If this isn't expected to actually happen, would it be better to have an `assert` here instead?

Reply:
In my list is to refactor this part and use the same interface for 2 latest tracers... just didn't want to mix it here

Reply:
forget about it.. i have to remove this part